src/main/java/net/dries007/tfc/world/river/RiverNoise.java

```diff
@@ -130,6 +228,63 @@ else if (y > SEA_LEVEL_Y - 8)
         };
     }
 
+    // Vertical cliff with 45 degree slopes above and below
+    public static RiverNoiseSampler talus(Seed seed)
+    {
+        return new RiverNoiseSampler() {
+
+            final Noise2D baseNoise = new OpenSimplex2D(seed.next()).octaves(4).spread(0.05f).scaled(-2.5f, 1.5f);
+            final Noise2D cliffHeightNoise = new OpenSimplex2D(seed.next()).octaves(2).spread(0.1f).scaled(3f, 8f);
+            final Noise2D distNoise = new OpenSimplex2D(seed.next()).octaves(4).spread(0.05f).scaled(-0.15f, 0.15f);
+
+            double height;
+
+            @Override
+            public double setColumnAndSampleHeight(RiverInfo info, int x, int z, double heightIn, double caveWeight, double thisWeight)
```

Comment:
trying to understand why the param thisWeight was added, I can't find it used? I may be wrong.

Reply:
Right now it's used exclusively by the terrace river blend type, and the use case is that the blending with cave rivers tacitly assumes that the only tall_canyon biomes border caves. Since I didn't find much success modifying the cave blending to work better with other blend types, it made sense to just have terrace biomes mimic tall canyons at edges